### PR TITLE
Add Atlas Based Segmentation (ABS) profile

### PR DESCRIPTION
**Pull request by GuillaumeTh**
In the input folder you need the wmparc and aparc+aseg in nifiti in the same space of the raw T1 image. 

Then just use ABS profile and run tractoflow.

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -1053,6 +1067,100 @@ process Register_T1 {
...
+
+    script:
+    """
+    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
+    export OMP_NUM_THREADS=1
+    export OPENBLAS_NUM_THREADS=1
+    mkdir wmparc_desikan/
+    mkdir wmparc_subcortical/
+    mkdir aparc+aseg_subcortical/
+    mrconvert $aparc aparc+aseg_int16.nii.gz -datatype int16 -force -nthreads 1
```
Comment (arnaudbore) [inline main.nf]: Please use: `scil_image_math.py`

File: main.nf [hunk 2]
```diff
@@ -1053,6 +1067,100 @@ process Register_T1 {
...
+                             wmparc_subcortical/*amygdala*\
+                             wmparc_subcortical/*accumbens*\
+                             wmparc_subcortical/*plexus*\
+                             mask_nuclei_m.nii.gz -f
+    scil_image_math.py union wmparc_subcortical/*-lateral-ventricle.nii.gz\
+                             wmparc_subcortical/*-inferior-lateral-ventricle.nii.gz\
+                             wmparc_subcortical/cerebrospinal-fluid.nii.gz\
+                             wmparc_subcortical/*th-ventricle.nii.gz\
+                             mask_csf_1_m.nii.gz -f
+    mrthreshold mask_wm_m.nii.gz ${sid}__mask_wm_bin.nii.gz -abs 0.1 -force -nthreads 1
```
Comment (arnaudbore) [inline main.nf]: `scil_image_math.py`

**Review by arnaudbore — COMMENTED** [review]
A couple of things to convert: mrtrix -> scilpy
Later, I will do the same for the rest of tractoflow.

File: main.nf [hunk 3]
```diff
@@ -1013,9 +1027,9 @@ process Register_T1 {
 
     output:
     set sid, "${sid}__t1_warped.nii.gz" into t1_for_seg
-    file "${sid}__output0GenericAffine.mat"
+    set sid, "${sid}__t1_warped.nii.gz", "${sid}__output0GenericAffine.mat",
+        "${sid}__output1InverseWarp.nii.gz" into t1_for_freesurfer_reg
     file "${sid}__output1InverseWarp.nii.gz"
-    file "${sid}__output1Warp.nii.gz"
```
Comment (liuzhenqi77) [inline main.nf:1018]: Hi, I'm wondering why `output1Warp` is removed from the file output? Thanks.

File: main.nf [hunk 4]
```diff
@@ -1053,6 +1067,100 @@ process Register_T1 {
...
+labels_for_reg
+    .join(t1_for_freesurfer_reg)
+    .set{labels_mat_for_reg}
+
+process Register_Freesurfer {
+    cpus 1
+
+    input:
+    set sid, file(aparc), file(wmparc), file(t1), file(affine),
+        file(warp) from labels_mat_for_reg
```
Comment (liuzhenqi77) [inline main.nf:1079]: Also, I'm wondering is this `warp` file `output1InverseWarp` or `output1Warp`?
In the older repo [here](https://github.com/scilus/tractoflow-ABS/blob/master/main.nf#L957-L962), it seems to be using `output1Warp`, but here is seems only `output1InverseWarp` is in `labels_mat_for_reg`? Thanks.

File: main.nf (hunk 3 above)
Comment (GuillaumeTh) [reply main.nf:1018]: Hi @liuzhenqi77 thanks for the comment. I will fix this.

File: main.nf (hunk 4 above)
Comment (GuillaumeTh) [reply main.nf:1079]: Thanks it is fixed now.

**Pull request merged**